README.md

```diff
@@ -1 +1,112 @@
+([source code](https://github.com/keycloak/keycloak)) in order to host a custom
+SAML-based Identity Provider (IdP) without having to run a server or rely on
+some external service. Other tools were investigated in JavaScript 
+([@node-saml/node-saml](https://github.com/node-saml/node-saml/),
+[xml-crypto](https://github.com/node-saml/xml-crypto/),
+[saml-idp](https://github.com/mcguinness/saml-idp), and
+[@node-saml/passport-saml](http://www.passportjs.org/packages/passport-saml/)),
+but they all seemed to be bad fits (node-saml explicitly is intended to be used
+on the Service Provider (SP) side, I was unable to generate a valid signature
+with xml-crypto, saml-idp had insecure dependencies, and passport-saml may
+actually work (I should definitely come back to
+[this](https://github.com/node-saml/passport-saml/blob/master/docs/xml-signing-example.js)
+especially), but I didn't look into it very hard after seeing that it was built
+to be used with [ExpressJS](https://expressjs.com/), especially knowing that
+this was only going to be backend code).
+
+In light of these issues, a dependable, open-source IdP was chosen, even though
+our office generally does not support Java. A 
+[medium article](https://neuw.medium.com/aws-connect-saml-based-identity-provider-using-keycloak-9b3e6d0111e6)
+about manually configuring exactly the type of Keycloak client needed here was
+followed, and after that client was verified to correctly federate sign-ons to
+AWS, an intensive live-debugging process allowed us to find a thin slice of ~500
+lines of code along with hardcoded stand-ins that could be used in this
+repository. The Keycloak code was copied, massaged, and adapted to be a
+standalone solution for generating those SAML Response documents used for
+logging in to AWS Connect.
+
+## Requirements and Deployment
+
+1. This is built in Java 17 using Gradle
+
+2. AWS (in the account where the target AWS Connect instance lives):
+   1. Create a SAML-based Identity Provider in IAM using the metadata document
+      [`src/main/resources/identity-provider-metadata.xml`](src/main/resources/identity-provider-metadata.xml)
+
+   2. Create a Policy in IAM which allows getting federation tokens for the
+      target Connect instance. Generic example:
+   ```json
+   {
+       "Version": "2012-10-17",
+        "Statement": [{
+            "Sid": "FederationForConnect",
+            "Effect": "Allow",
+            "Action": "connect:GetFederationToken",
+            "Resource": "*",
+            "Condition": {
+                "StringEquals": {
+                    "connect:InstanceId": "<TARGET_CONNECT_INSTANCE_ID>"
+                }
+            }
+        }]
+   }
+   ```
+
+   3. Create a Role:
+      1. For _SAML 2.0 federation_
+      2. Using the above-created Identity Provider
+      3. With "_Allow programmatic and AWS Management Console access_"
+      4. With the above-created Policy
+
+   4. Upload key-secrets to AWS Systems Manager -> Parameter Store:
```

Comment:
This should be a separate point not connected to #2, it does not need to live in the AWS Account where the Connect instance lives, it just needs to be in the AWS Account where this code is deployed.